src/Microsoft.AspNet.Mvc.ViewFeatures/Buffer/ViewBuffer.cs

```diff
@@ -35,17 +37,14 @@ public ViewBuffer(IViewBufferScope bufferScope, string name)
 
             _bufferScope = bufferScope;
             _name = name;
+
+            Pages = new List<ViewBufferPage>();
```

Comment:
I changed this to track the count per-page. It's not a lot to allocate, and it makes this code so much simpler.